src/full/Agda/Syntax/Common.hs

```diff
@@ -1254,6 +1254,11 @@ data Annotation = Annotation
   { annLock :: Lock
     -- ^ Fitch-style dependent right adjoints.
     --   See Modal Dependent Type Theory and Dependent Right Adjoints, arXiv:1804.05236.
+  , annFinite :: !Bool
```

Comment:
If I read you right, the reason to move this from `Dom` to `ArgInfo` was to have it also in the abstract syntax.  Currently, the abstract syntax has no `Dom`, but:
```haskell
data TypedBinding
  = TBind Range TacticAttr (List1 (NamedArg Binder)) Type
  | ...
```
Would it make sense to turn the `TacticAttr` field to a record containing also the `Finite` flag?  I am a bit worried about blowing up `ArgInfo` unnecessarily, because it is part of _every argument and binder_.  A unused field there would increase the memory footprint a bit (maybe quite a bit?).  If possible, I'd appreciate a more economic placement.

(More radically, refactoring the abstract syntax to have `Dom` would also be thinkable...)

Reply:
Right: the finiteness annotation has to be present in the abstract syntax to differentiate the pretty-printed normal forms of `Partial φ A` and `.(IsOne φ) → A`.

I don't think the size of `ArgInfo` has been significantly affected, though: `annFinite` is part of the `Annotation` type, which was already pointed to by `ArgInfo` before, and since the field isn't strict, it doesn't get unboxed (though maybe it _should_ be made strict/unboxed?)

```haskell
data ArgInfo = ArgInfo
  { argInfoHiding        :: Hiding
  , argInfoModality      :: Modality
  , argInfoOrigin        :: Origin
  , argInfoFreeVariables :: FreeVariables
  , argInfoAnnotation    :: Annotation
    -- ^ Sometimes we want a different kind of binder/pi-type, without it
    --   supporting any of the @Modality@ interface.
  } deriving (Data, Eq, Ord, Show)
```

Nonetheless I think it makes sense to store it next to the `TacticAttr`, since arguments in elimination spines shouldn't particularly care about the finiteness of the function they're being applied to.